widget/button.go

```diff
@@ -655,73 +674,108 @@ func (b *Button) initText() {
 	b.container.AddChild(b.text)
 
 	b.autoUpdateTextAndGraphic = true
-
 }
 
 func (b *Button) createWidget() {
 	b.widget = NewWidget(append(b.widgetOpts, []WidgetOpt{
 		WidgetOpts.CursorEnterHandler(func(args *WidgetCursorEnterEventArgs) {
-			if !b.widget.Disabled {
-				b.hovering = true
+			if b.mask == nil {
+				if !b.widget.Disabled {
+					b.hovering = true
+				}
+				if b.hovering {
+					b.CursorEnteredEvent.Fire(&ButtonHoverEventArgs{
+						Button:  b,
+						Entered: true,
+						OffsetX: args.OffsetX,
+						OffsetY: args.OffsetY,
+						DiffX:   0,
+						DiffY:   0,
+					})
+				}
 			}
-			b.CursorEnteredEvent.Fire(&ButtonHoverEventArgs{
-				Button:  b,
-				Entered: true,
-				OffsetX: args.OffsetX,
-				OffsetY: args.OffsetY,
-				DiffX:   0,
-				DiffY:   0,
-			})
 		}),
 
 		WidgetOpts.CursorMoveHandler(func(args *WidgetCursorMoveEventArgs) {
-			if !b.widget.Disabled {
-				b.hovering = true
+			if b.onMask(args.OffsetX, args.OffsetY) {
+				if !b.hovering {
+					b.CursorEnteredEvent.Fire(&ButtonHoverEventArgs{
+						Button:  b,
+						Entered: true,
+						OffsetX: args.OffsetX,
+						OffsetY: args.OffsetY,
+						DiffX:   0,
+						DiffY:   0,
+					})
+				}
+				if !b.widget.Disabled {
+					b.hovering = true
+				}
+				b.CursorMovedEvent.Fire(&ButtonHoverEventArgs{
+					Button:  b,
+					Entered: false,
+					OffsetX: args.OffsetX,
+					OffsetY: args.OffsetY,
+					DiffX:   args.DiffX,
+					DiffY:   args.DiffY,
+				})
+			} else {
+				if b.hovering {
+					b.hovering = false
+					b.CursorExitedEvent.Fire(&ButtonHoverEventArgs{
+						Button:  b,
+						Entered: false,
+						OffsetX: args.OffsetX,
+						OffsetY: args.OffsetY,
+						DiffX:   0,
+						DiffY:   0,
+					})
+				}
 			}
-			b.CursorMovedEvent.Fire(&ButtonHoverEventArgs{
-				Button:  b,
-				Entered: false,
-				OffsetX: args.OffsetX,
-				OffsetY: args.OffsetY,
-				DiffX:   args.DiffX,
-				DiffY:   args.DiffY,
-			})
 		}),
 
 		WidgetOpts.CursorExitHandler(func(args *WidgetCursorExitEventArgs) {
-			b.hovering = false
-			b.CursorExitedEvent.Fire(&ButtonHoverEventArgs{
-				Button:  b,
-				Entered: false,
-				OffsetX: args.OffsetX,
-				OffsetY: args.OffsetY,
-				DiffX:   0,
-				DiffY:   0,
-			})
-		}),
-
-		WidgetOpts.MouseButtonPressedHandler(func(args *WidgetMouseButtonPressedEventArgs) {
-			if !b.widget.Disabled && args.Button == ebiten.MouseButtonLeft {
-				b.pressing = true
-
-				b.PressedEvent.Fire(&ButtonPressedEventArgs{
+			if b.hovering || b.mask == nil {
```

Comment:
I dont think this will ever run if ignoreTranparentPixels is false

Reply:
Oh thought the same, but actually it does.
Here is how:

1. Press & hold anywhere within the mask
2. While pressed move the mouse cursor outside of the widget or even the game window, where mouse move event of  the widget itself is not triggered